draft-ietf-tsvwg-sctp-dtls-chunk.md

```diff
@@ -1217,31 +1217,438 @@ some extension of the socket API.
+``SCTP_COOKIE_ECHOED``.
+
+### Set or Get Protection Enforcement (``SCTP_DTLS_ENFORCE_PROTECTION``)
+
+Enabling this socket option on an SCTP endpoint enforces that only
+protected SCTP packets are processed anymore. All received packets with the
+first chunk not being an INIT chunk, INIT ACK chunk, or DTLS chunk will be
+silently discarded.
+
+The following structure is used as the ``option_value``:
+
+~~~ c
+struct sctp_assoc_value {
+	sctp_assoc_t assoc_id;
+	uint32_t assoc_value;
+};
+~~~
+
+``assoc_id``:
+: This parameter is ignored for one-to-one style sockets.
+  For one-to-many style sockets, this parameter indicates upon which
+  SCTP association the caller is performing the action.
+  It is an error to use ``SCTP_{FUTURE|CURRENT|ALL}_ASSOC``.
+
+``assoc_value``:
+  The value `0` represents that the option is off, any other value represents
+  that the option is on.
+
+This socket option is off by default on any SCTP endpoint.
+Once protection has been enforced by enabling this socket option on an
+SCTP endpoint, it cannot be disabled again.
+Whether protection has been enforced on an SCTP endpoint can be queried
+in any state other than ``SCTP_CLOSED``.
+Protection can be enforced in any state other than ``SCTP_CLOSED``,
+``SCTP_COOKIE_WAIT`` and ``SCTP_COOKIE_ECHOED``.
+
+### Get Statistic Counters (``SCTP_DTLS_STATS``)
+
+This socket options allows to get various statistic counters for a
+specific SCTP endpoint.
+
+The following structure is used as the ``option_value``:
+
+~~~ c
+struct sctp_dtls_stats {
+	sctp_assoc_t sds_assoc_id;
+	uint32_t sds_dropped_unprotected;
+	uint32_t sds_aead_failures;
+	/* There will be added more fields before the WGLC. */
+	/* There might be additional platform specific counters. */
+};
+~~~
+
+``sds_assoc_id``:
+: This parameter is ignored for one-to-one style sockets.
+  For one-to-many style sockets, this parameter indicates upon which
+  SCTP association the caller is performing the action.
+  It is an error to use ``SCTP_{FUTURE|CURRENT|ALL}_ASSOC``.
+
+``sds_dropped_unprotected``:
+: The number of of unprotected packets received for this SCTP endpoint after
+  protection was enforced.
+
+``sds_aead_failures``:
+: The number of of AEAD failures when processing received DTLS chunks.
+
+### Get or Set the Replay Protection Window Size (``SCTP_DTLS_REPLAY_WINDOW``)
+
+This socket option can be used to configure the replay protection for SCTP
+endpoints.
+
+The following structure is used as the ``option_value``:
+
+~~~ c
+struct sctp_assoc_value {
+	sctp_assoc_t assoc_id;
+	uint32_t assoc_value;
+};
+~~~
+
+``assoc_id``:
+: This parameter is ignored for one-to-one style sockets.
+  For one-to-many style sockets, this parameter indicates upon which
+  SCTP association the caller is performing the action.
+  It is an error to use ``SCTP_{CURRENT|ALL}_ASSOC``.
+
+``assoc_value``:
+  The size of the replay protection window.
+  A value of ``0`` indicates that the replay protection is disabled.
```

Comment:
Based on my discussions with @emanjon making it possible to disable replay protection might be quite bad for the security properties of the association, at least as long as one are using for example AES-GCM as cipher. It might be less of a problem for other ciphers that doesn't have weakness to authentication replay. So I think this option needs to be have a security consideration discussion if included.

Reply:
Had a discussion with @hannestschofenig : they plan to make replay protection mandatory in the DTLS 1.3bis. So assume in this document that it is there. We need to have a discussion about the size in the DTLS considerations.
I removed the sentence that you can disable the replay protection.